### PR TITLE
🎉 Source Postres: Added new comprehensive data type tests

### LINKED ISSUE #3795: Source Postgres: create comprehensive data type tests
## Tell us about the problem you're trying to solve
Same as https://github.com/airbytehq/airbyte/issues/3562 for Postgres
postgres version: postgres:13-alpine
https://www.postgresql.org/docs/13/datatype.html#:~:text=The%20following%20types%20(or%20spellings,without%20time%20zone)%2C%20xml%20.

TODO:

1. Read documentation and prepare tricky use cases for each type of data. 
2. Implement comprehensive test. 
3. Prepare pull request, pass all checks. 

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200368060087111/1200413427027728) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by etsybaev**
## What
The current test coverage level was too low.

## How
 Added new comprehensive data type tests. All new issues have been described in new follow up ticket https://github.com/airbytehq/airbyte/issues/4071

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in output spec
- [x] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [x] Code reviews completed
- [x] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Documentation updated 
    - [ ] README
    - [ ] CHANGELOG.md
    - [ ] Reference docs in the `docs/integrations/` directory.
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [x] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [x] No major blockers
- [x] PR merged into master branch
- [x] Follow up tickets have been created
- [x] Associated tickets have been closed & stakeholders notified
</p>
</details>

### CODE REVIEW

**Review by yaroslav-hrytsaienko — COMMENTED** [review]
What about the documentation?

**Review by yaroslav-hrytsaienko — APPROVED** [review]

File: airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/CdcPostgresSourceComprehensiveTest.java [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+                "128.1.0.0/16", "2001:4f8:3:ba::/64")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("date")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .addInsertValues("null")
+            .addNullExpectedValue()
+            .build());
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/CdcPostgresSourceComprehensiveTest.java:239]: Nit: this case is included in the one below. So it seems redundant.

File: airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/PostresSourceComprehensiveTest.java [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+                "128.1.0.0/16", "2001:4f8:3:ba::/64")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("date")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .addInsertValues("null")
+            .addNullExpectedValue()
+            .build());
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/PostresSourceComprehensiveTest.java:229]: Same here.

**Review by tuliren — APPROVED** [review]
Looks good! Only one comment about a redundant test case for `date`.

File: airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/PostresSourceComprehensiveTest.java (hunk 2 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/PostresSourceComprehensiveTest.java]: removed redundant, thanks

File: airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/CdcPostgresSourceComprehensiveTest.java (hunk 1 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/CdcPostgresSourceComprehensiveTest.java]: removed redundant, thanks

**Comment by cgardens** [comment]
This is cool. I like it.

**Pull request merged**